chpl/chpl-service/src/test/java/gov/healthit/chpl/manager/auth/UserManagerTest.java

```diff
@@ -0,0 +1,450 @@
+
+        Mockito.doNothing()
+                .when(activityManager)
+                .addActivity(ArgumentMatchers.any(), ArgumentMatchers.anyLong(), ArgumentMatchers.anyString(),
+                        ArgumentMatchers.any(), ArgumentMatchers.any());
+
+        Mockito.doNothing()
+                .when(userResetTokenDAO)
+                .deletePreviousUserTokens(ArgumentMatchers.anyLong());
+        Mockito.when(userResetTokenDAO.create(ArgumentMatchers.anyString(), ArgumentMatchers.anyLong()))
+                .thenReturn(UserResetTokenDTO.builder().build());
+        Mockito.when(userResetTokenDAO.findByAuthToken(ArgumentMatchers.anyString()))
+                .thenReturn(UserResetTokenDTO.builder()
+                        .creationDate(new Date())
+                        .build());
+    }
+
+    @Test
+    public void create_GoodData_Success() throws UserCreationException {
+
+        UserDTO user = getUserDTO(1L);
+        UserManager userManager = new UserManager(null, userDAO, null, bCryptPasswordEncoder, mutableAclService, null, null);
+
+        UserDTO newUser = userManager.create(user, "@ThisShouldBeAStrongPassword1!");
+
+        assertNotNull(newUser);
+    }
+
+    @Test(expected = UserCreationException.class)
+    public void create_WeakPassword_UserCreationException() throws UserCreationException {
+        Mockito.when(userDAO.create(ArgumentMatchers.any(UserDTO.class), ArgumentMatchers.anyString()))
+                .thenThrow(new UserCreationException());
+
+        UserDTO user = getUserDTO(1L);
+
+        UserManager userManager = new UserManager(null, userDAO, null, bCryptPasswordEncoder, mutableAclService, null, null);
+
+        userManager.create(user, "@ThisShouldBeAStrongPassword1!");
+
+        fail();
+    }
+
+    @Test
+    public void update_GoodData_SuccessAndActivityWritten()
+            throws UserRetrievalException, JsonProcessingException, EntityCreationException, EntityRetrievalException,
+            ValidationException {
+
+        UserManager userManager = new UserManager(null, userDAO, null, null, null, null, activityManager);
+
+        UserDTO user = getUserDTO(1L);
+        UserDTO updatedUser = userManager.update(user);
+
+        assertNotNull(updatedUser);
+
+        Mockito.verify(activityManager).addActivity(ArgumentMatchers.any(), ArgumentMatchers.anyLong(),
+                ArgumentMatchers.anyString(), ArgumentMatchers.any(), ArgumentMatchers.any());
+    }
+
+    @Test(expected = ValidationException.class)
+    public void update_MissingFullName_ValidationExceptionThrown()
+            throws UserRetrievalException, JsonProcessingException, EntityCreationException, EntityRetrievalException,
+            ValidationException {
+
+        UserManager userManager = new UserManager(null, null, null, null, null, errorMessageUtil, null);
+
+        UserDTO user = getUserDTO(1L);
+        user.setFullName("");
+        userManager.update(user);
```

Comment:
Just so I'm clear. If this method does not throw an exception (which is expected by the annotation) then the test will go to the next command and fail the test, right?